src/main/java/org/terasology/climateConditions/FrostbiteSystem.java

```diff
@@ -57,18 +55,18 @@
     private static final int healthDecreaseAmount = 15;
     private Random random = new FastRandom();
 
-    @ReceiveEvent(components = {HypothermiaComponent.class})
-    public void onHypothermia(OnAddedComponent event, EntityRef player) {
+
+    public void applyFrostbite(EntityRef player) {
```

Comment:
Why is this method public? If it needs to be called by other systems (outside of this module) this should rather be handled via the ECS (e.g., use and event instead of a direct method call).

Reply:
Commented on Discord as well, but yeah I agree. IMHO either private methods used just within the System class itself, or straight up event handler methods. If anything in a System gets used externally it would be the sort of thing you'd `@Share` in a utility-focused System, like something related to inventory :-)

Speaking of: I never really thought about whether event handler methods themselves generally end up public and if you _could_ call them directly as well 🤔 I guess that would at least be weird since you'd have to pass in parameters. I'd imagine only the ES itself should ever call event handlers.